analysis/README.md

```diff
@@ -0,0 +1,26 @@
+# Analysis Library and Binary
+
+This subfolder builds a private command line binary used by the plugin to power a few functionalities such as jump to definition, hover and autocomplete.
+
+The binary reads the `.cmt` and `.cmti` files and analyses them.
+
+For installation & build instructions, see the main CONTRIBUTING.md.
+
+## Overview
+
+See main CONTRIBUTING.md's repo structure. Additionally, `examples/` is a convenience debugging repo. Check out `test.sh` (invoked through `make test`) to see the snapshots testing workflow stored in `tests/`.
```

Comment:
I guess changes to CONTRIBUTING are coming in a separate PR.

Reply:
Yes, still some stuff to clean up, and also to move/add to the root Makefile.